pkg/controller/migplan/pvlist.go

```diff
@@ -248,15 +258,17 @@ func getMappedNameForPVC(pvcRef *core.ObjectReference, podList []core.Pod, plan
 	// if plan is for storage conversion, create a new unique name for the destination PVC
 	if isStorageConversionPlan(plan) &&
 		(existingPVC == nil || existingPVC.GetSourceName() == existingPVC.GetTargetName()) {
-		// we cannot append a prefix when pvcName itself is 252 characters or more because:
+		// we cannot append a suffix when pvcName itself is 247 characters or more because:
 		// 1. total length of new pvc name cannot exceed 253 characters
-		// 2. we append a '-' char before prefix and pvc names cannot end with '-'
-		if len(pvcName) > 251 {
+		// 2. we append '-mig-' char before prefix and pvc names cannot end with '-'
+		if len(pvcName) > 247 {
```

Comment:
number 247 doesn't sound right, shouldn't it consider the variable length `plan.GetSuffix()`?

Reply:
247 ensures we get `mig-x` at worst, this is the same as before where we could get `-n` instead of `-new` if we had 251 characters.

Reply:
So before we added the suffix `-new`, and now the suffix is `-mig-xxxx` with xxxx being a random 4 characters. If you check the unit tests, I ensure there that we are not creating names that end with a `-`. The 247 is from checking with the unit tests that nothing illegal is generated.